Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerController.swift

```diff
@@ -9,35 +9,53 @@ public protocol SwiftPackageManagerControlling {
     /// - Parameter path: Directory where the `Package.swift` is defined.
     func resolve(at path: AbsolutePath) throws
 
-    /// Update package dependencies
+    /// Updates package dependencies.
     /// - Parameter path: Directory where the `Package.swift` is defined.
     func update(at path: AbsolutePath) throws
+
+    /// Sets tools version of package to the given value.
+    /// - Parameter path: Directory where the `Package.swift` is defined.
+    /// - Parameter version: Version of tools. When `nil` then the environment’s version will be set.
+    func setToolsVersion(at path: AbsolutePath, to version: String?) throws
 }
 
 public final class SwiftPackageManagerController: SwiftPackageManagerControlling {
     public init() {}
 
     public func resolve(at path: AbsolutePath) throws {
-        let command = buildSwiftPackageCommand(packagePath: path, subcommand: "resolve")
+        let command = buildSwiftPackageCommand(packagePath: path, subcommand: ["resolve"])
 
         try System.shared.run(command)
     }
 
     public func update(at path: AbsolutePath) throws {
-        let command = buildSwiftPackageCommand(packagePath: path, subcommand: "update")
+        let command = buildSwiftPackageCommand(packagePath: path, subcommand: ["update"])
+
+        try System.shared.run(command)
+    }
+
+    public func setToolsVersion(at path: AbsolutePath, to version: String?) throws {
```

Comment:
This is a bit misleading. As a user of the `swiftPackageManager`, you call a method, `setToolsVersion`, and it turns out the method is building a Swift package. What do you think about the following:
- Remove `setToolsVersion` from the protocol.
- Turn this method into `func toolsVersionArguments(configVersion: String?) -> [String]`
- Update the methods in this controller to add the extra arguments provided by this method.

Is there an scenario where you don't want the `update` and `resolve` methods in this class to use those arguments?

Reply:
>  you call a method, setToolsVersion, and it turns out the method is building a Swift package. 

I don't understand.

This methods does not build a Swift package. 

It calls 

```bash
swift package --package-path [PATH] tools-version --set-current
```
or

```bash
swift package --package-path [PATH] tools-version --set [Version]
```

It there is no `Package.swift` at [PATH] then it ends with:

<img width="738" alt="Zrzut ekranu 2021-05-25 o 21 49 48" src="https://user-images.githubusercontent.com/4774319/119559558-21d72980-bda3-11eb-9f76-0f4e171b6c6b.png">

If `Package.swift` exists at [PATH] then the `swift-tools-version` updates:

<img width="806" alt="Zrzut ekranu 2021-05-25 o 21 50 43" src="https://user-images.githubusercontent.com/4774319/119559656-429f7f00-bda3-11eb-878c-ee6cd1bf62ae.png">

<img width="917" alt="Zrzut ekranu 2021-05-25 o 21 50 58" src="https://user-images.githubusercontent.com/4774319/119559690-4cc17d80-bda3-11eb-9d63-1b6c6db21217.png">

So `setToolsVersion(at)` set `swift-tools-version` at the given path. Nothing more. 🤔 

Please correct me if I miss something.

Reply:
<img width="552" alt="Zrzut ekranu 2021-05-25 o 21 56 58" src="https://user-images.githubusercontent.com/4774319/119560347-218b5e00-bda4-11eb-8a0f-1955e59e45a7.png">

Reply:
Thanks for clarifying it. I misunderstood the usage. I'm very surprised Apple went down the path of adding a command instead of allowing controlling that through arguments.